src/pcg/corridor_generator.cpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+#include <sstream>
+#include <cassert>
+#include "src/core/components/graphics_component.h"
+#include "src/core/components/transformation_component.h"
+#include "src/core/components/description_component.h"
+#include "src/core/scene.h"
+#include "src/core/entity.h"
+#include "src/pcg/corridor_generator.h"
+
+/*
+Corridors are connected using "midpoint" i.e. a point that is exactly between two centres
+of the rooms. A corridor always is constructed as 3 entities. Horizontal connection example:
+
+ooo
+ooo
+oAo
+ o
+ BoMoC
+     o
+    oDo
+    ooo
+    ooo
+
+M - midpoint
+A - first_connecting_point
+B - second_connecting_point
+C - third_connecting_point
+D - fourth_connecting_point
+*/
+
+
+std::vector<BoundingBox> CorridorGenerator::calculate_corridor_boxes() {
```

Comment:
This method is vastly too long. Split it to separate functions, including extracting more advanced conditions, like `std::abs(first_middle.x - second_middle.x) < std::abs(first_middle.y - second_middle.y`